File: android-native/FretboardHeatmap/app/src/main/java/com/example/fretboardheatmap/BottomMenuLabels.kt
package com.example.fretboardheatmap

object BottomMenuLabels {

    /**
     * Provides the bottom label menus
     * @param root root note choice
     * @param topMenu top menu choice
     * @param dropdownChoice the right bottom menu labels
     */
    fun getLabels(
        root: String,
        topMenu: TopMenuChoice?,
        dropdownChoice: String
    ): String {
        return if (minorKeySelected(topMenu, dropdownChoice)) {
            "${root}m"
        } else {
            root
        }
    }

    /**
     * Prepare bottom menu minor labels
     * @param topMenu top menu choice
     * @param dropdownChoice top menu dropdown choice
     */
    private fun minorKeySelected(topMenu: TopMenuChoice?, dropdownChoice: String): Boolean {
        if (topMenu == null) return false

        val isMinor = dropdownChoice.startsWith("Min", ignoreCase = true)

        return isMinor
    }
}

File: ios-native/FretboardHeatmap/BottomMenuLabels.swift
//
//  BottomMenuLabels.swift
//  FretboardHeatmap
//
//  Created by Sean Avery Suguitan on 2026-02-09.
//

import Foundation

// Provides the bottom label menus
struct BottomMenuLabels {
    
    /// Prepare bottom menu labels
    /// - Parameters:
    ///   - root: Root note choice
    ///   - topMenu: Top menu choice
    ///   - dropdownChoice: Top menu dropdown choice
    /// - Returns: The right bottom menu labels
    static func getLabels (
        for root: String,
        topMenu: menuChoice?,
        dropdownChoice: String
    ) -> String {
        if minorKeySelected(topMenu: topMenu, dropdownChoice: dropdownChoice) {
            return "\(root)m"
        }
        return root
    }
    
    /// Prepare bottom menu minor labels
    /// - Parameters:
    ///   - topMenu: Top menu choice
    ///   - dropdownChoice: Top menu dropdown choice
    /// - Returns: Minor labels (if needed)
    private static func minorKeySelected(topMenu: menuChoice?, dropdownChoice: String) -> Bool {
            guard let menu = topMenu else { return false }
            
            let isMinor = dropdownChoice.hasPrefix("Min")
            
            return (menu == .scales || menu == .chords) && isMinor
        }
}
